timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -241,11 +243,31 @@ def fit(
         model: AbstractTimeSeriesModel
             The fitted model object
         """
+        self.initialize(**kwargs)
+        self.target_scaler = self._get_target_scaler()
```

Comment:
I see two reasons for this choice:
1. `target_scaler` is a special kind of preprocessing, since we also need to apply the `inverse_transform` on the predictions. In contrast, `preprocess` only modifies the data before it's consumed by the model.
2. `self.preprocess` contains model-specific preprocessing logic that can be modified by the subclasses. By keeping the `target_scaler` out of it, we don't need to rely on the person implementing the model to remember using the `target_scaler` appropriately.

Reply:
This seems plausible to me, thanks a lot for clarifying!